gobblin-data-management/src/main/java/gobblin/data/management/retention/DatasetCleaner.java

```diff
@@ -53,6 +71,10 @@ public DatasetCleaner(FileSystem fs, Properties props) throws IOException {
     } catch (InvocationTargetException exception) {
       throw new IOException(exception);
     }
+    this.executor =
+        Executors
+            .newFixedThreadPool(Integer.parseInt(props.getProperty(DATASET_CLEAN_THREAD_POOL_SIZE,
```

Comment:
Using a ThreadPoolExecutor with a high (say, 1000) thread limit and a core pool size of 0 should be fine. The point is we should not be using # of threads to control performance.